index.html

```diff
@@ -62,6 +62,16 @@
               );
             }
           }
+        },
+        function checkPracticeIds(_conf, _doc, utils) {
+          // Ensure every principle has a non-generated ID.
+          for (const practice of document.querySelectorAll(".practicelab")) {
+            if (!practice.id) {
+              utils.showWarning(
+                `Principle should be assigned an id.`, { elements: [practice] }
+              );
+            }
+          }
```

Comment:
This might also work? #codeshedding =)

```suggestion
          document.querySelectorAll(".practicelab:not([id])").forEach(practice => {
              utils.showWarning(
                `Principle should be assigned an id.`, { elements: [practice] }
              );
          }
```